### PR TITLE
Initial rmf-codegen generated codebase

### PR DESCRIPTION
**Pull request by mvantellingen**
New codebase generated on https://github.com/labd/rmf-codegen/tree/feature/python

Todo:
 - [x] Handle regex query parameters
 - [x] Validate image upload calls
 - [x] Make sure that subtypes are returned from supertype deserialization
 - [x] Decide if we want to convert camel case request builder properties to snake case

To be done later:
 - [ ] Convert more unittests to new client (later s
 - [ ] Generate error handling based on raml

### CODE REVIEW

File: tests/platform/test_service_cart.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+from commercetools.platform import models
+
+
+def test_cart_get_by_id(ct_platform_client: Client, cart_draft):
+    cart = (
+        ct_platform_client
+        .withProjectKey("test")
+        .carts()
+        .post(cart_draft))
+
```
Comment (mvantellingen) [inline tests/platform/test_service_cart.py:15]: Example for new `client.carts.create()` call

File: tests/platform/test_service_cart.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+def test_update_actions(commercetools_api, ct_platform_client, cart_draft):
+    cart = ct_platform_client.withProjectKey("test").carts().post(cart_draft)
+
+    payment_reference = models.PaymentReference(id=str(uuid.uuid4()))
+    cart = ct_platform_client.withProjectKey("test").carts().withId(cart.id).post(
+        models.CartUpdate(
+            version=cart.version,
+            actions=[models.CartAddPaymentAction(payment=payment_reference)],
+        )
+    )
```
Comment (mvantellingen) [inline tests/platform/test_service_cart.py:68]: example of new `client.carts.update_by_id()` call

File: tests/platform/test_service_cart.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            version=cart.version,
+            actions=[
+                models.CartSetCustomFieldAction(name="foo1", value="bar"),
+                models.CartSetCustomFieldAction(name="foo2", value=["bar"]),
+                models.CartSetCustomFieldAction(name="foo3", value=False),
+            ],
+        )
+    )
+
+    cart = ct_platform_client.withProjectKey("test").carts().withId(cart.id).get()
```
Comment (mvantellingen) [inline tests/platform/test_service_cart.py:121]: example of new `client.carts.get_by_id()` call

**Comment by davidweterings** [comment]
Very nice :^), looked through various file and don't see a lot of issues.

- We can remove the 'from commercetools.helpers import RemoveEmptyValuesMixin' import?
- Is  there a way we can fix the typing for the `response_class`? 

image

Looks that way on both python 3.7 and 3.9

**Comment by mvantellingen** [comment]
> Very nice :^), looked through various file and don't see a lot of issues.
> 
> * We can remove the 'from commercetools.helpers import RemoveEmptyValuesMixin' import?
Oops :-)

> * Is  there a way we can fix the typing for the `response_class`?

So i initially created it using the typing.Protocol() since we just require a `deserialize` classmethod and a `serialize` method, but that is only working with 3.8. Will test `typing.Type[Model]`

> 
> image
> 
> Looks that way on both python 3.7 and 3.9

**Pull request merged**